modules/reindex/src/test/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSourceTests.java

```diff
@@ -88,6 +88,8 @@
 
     private final Queue<ScrollableHitSource.AsyncResponse> responseQueue = new LinkedBlockingQueue<>();
 
+    private final Queue<Throwable> failureQueue = new LinkedBlockingQueue<>();
```

Comment:
Can we add a `@After` that verifies that the `failureQueue` is empty? Otherwise, we could risk a request that fails without noticing it?

Reply:
++ done